Promise/Promise.swift

```diff
@@ -188,43 +228,65 @@ public final class Promise<Value> {
         })
     }
     
+    /// Handles the completion of a Promise.
+    /// - Parameters:
+    ///   - queue: Optional; queue to perform this work on.
+    ///            Defaults to the main queue.
+    ///   - onFulfilled: Work to perform if the Promise is fulfilled (completes with a value)
+    ///   - onRejected: Work to perform if the Promise is rejected (completes with an error)
+    /// - Returns: A discardable instance of this promise that can be used for further chaining.
     @discardableResult
     public func then(on queue: ExecutionContext = DispatchQueue.main, _ onFulfilled: @escaping (Value) -> Void, _ onRejected: @escaping (Error) -> Void = { _ in }) -> Promise<Value> {
         addCallbacks(on: queue, onFulfilled: onFulfilled, onRejected: onRejected)
         return self
     }
     
+    /// Catches an error if this promise chain is rejected and performs some work.
+    /// - Parameters:
+    ///   - queue: Optional; queue to perform this work on.
+    ///            Defaults to the main queue.
+    ///   - onRejected: Work to perform if this promise chain is rejected.
+    /// - Returns: A discardable instance of this promise that can be used for further chaining.
     @discardableResult
     public func `catch`(on queue: ExecutionContext = DispatchQueue.main, _ onRejected: @escaping (Error) -> Void) -> Promise<Value> {
         return then(on: queue, { _ in }, onRejected)
     }
     
+    /// Rejects this Promise (completes it with an `Error`).
+    /// - Parameter error: Error to reject with
     public func reject(_ error: Error) {
         updateState(.rejected(error: error))
     }
     
+    /// Fulfills this Promise (completes it succesfully with an instance of `Value`).
+    /// - Parameter value: Instance of `Value` to fulfill with
     public func fulfill(_ value: Value) {
         updateState(.fulfilled(value: value))
     }
     
+    /// Gets a flag indicating if this promise is still pending.
```

Comment:
```suggestion
    /// A flag indicating if the promise is still pending.
```

Reply:
@khanlou Personal preference: I like prefixing `var`s with "Gets" or "Gets or sets" to distinguish between those that are readonly and those that are read/write.

Not a hill I'll die on though.